librad/src/git/mod.rs

```diff
@@ -138,35 +138,34 @@ impl GitProject {
     pub fn init(
         paths: &Paths,
         key: &device::Key,
-        // TODO: Should accept a `ProfileRef`, but then we need to get hold of `nick`
-        profile: &meta::UserProfile,
         sources: &git2::Repository,
+        metadata: meta::Project,
+        founder: meta::Contributor,
     ) -> Result<ProjectId, Error> {
-        let project_name = {
-            let src_path = sources.path();
-            let file_name = if src_path.ends_with(".git") {
-                src_path.parent().and_then(|parent| parent.file_name())
-            } else {
-                src_path.file_name()
-            };
-            file_name
-                .map(|f| f.to_string_lossy().to_string())
-                .ok_or(Error::NeedProjectDir)
-        }?;
-
-        let mut pgp_key = key.clone().into_pgp(&profile.nick)?;
+        // TODO: resolve URL ref iff rad://
```

Comment:
```suggestion
        // TODO: resolve URL ref if rad://
```

Reply:
`iff = if and only if` :smartass_emoji:

Reply:
The reason I wrote that is that, in theory, one can also use any other URL -- but then we need to go out to the internet to fetch it. That sucks. 

Perhaps we can create a separate `Contributor` type which forces the caller to resolve the profile, and also specify if it should be stored inline or as a URL.

Reply:
:exploding_head: